Mathlib/Analysis/CStarAlgebra/CStarMatrix.lean

```diff
@@ -0,0 +1,489 @@
+    change ‖toCLM A M‖ = 0 ↔ M = 0
+    rw [norm_eq_zero]
+    exact toCLM_eq_zero_iff
+
+open CStarModule in
+lemma norm_entry_le_norm [DecidableEq n] [DecidableEq m] {M : CStarMatrix m n A} {i : m} {j : n} :
+    ‖M i j‖ ≤ ‖M‖ := by
+  have hmain : ‖M i j‖ ^ 3 * ‖M i j‖ ≤ ‖M i j‖ ^ 3 * ‖M‖ := calc
+        ‖M i j‖ ^ 4 = (‖M i j‖ * ‖M i j‖) * (‖M i j‖ * ‖M i j‖) := by ring
+        _ = ‖star (M i j * star (M i j)) * (M i j * star (M i j))‖ := by
+                rw [CStarRing.norm_star_mul_self, CStarRing.norm_self_mul_star]
+        _ = ‖⟪(WithCStarModule.equiv _).symm (Pi.single i (M i j * star (M i j))),
+                  toCLM A M ((WithCStarModule.equiv _).symm <| Pi.single j (star (M i j)))⟫_A‖ := by
+                simp only [star_mul, star_star, mul_assoc, WithCStarModule.inner_single_left,
+                  WithCStarModule.inner_def]
+                rw [toCLM_apply_single]
+                congr
+        _ ≤ ‖(WithCStarModule.equiv (m → A)).symm (Pi.single i (M i j * star (M i j)))‖
+                  * ‖toCLM A M ((WithCStarModule.equiv _).symm <| Pi.single j (star (M i j)))‖ :=
+                norm_inner_le (m →C⋆ A)
+        _ ≤ ‖M i j * star (M i j)‖ * ‖toCLM A M‖
+                  * ‖(WithCStarModule.equiv _).symm <| Pi.single j (star (M i j))‖ := by
+                rw [mul_assoc]
+                gcongr
+                · rw [WithCStarModule.norm_single]
+                · exact ContinuousLinearMap.le_opNorm (toCLM A M) _
+        _ = ‖M i j‖ ^ 2 * ‖M‖ * ‖M i j‖ := by
+                congr
+                · rw [CStarRing.norm_self_mul_star, pow_two]
+                · simp
+        _ = ‖M i j‖ ^ 3 * ‖M‖ := by ring
+  by_cases htriv : M i j = 0
+  · rw [htriv, norm_zero]
+    change 0 ≤ ‖toCLM A M‖
+    simp
+  · have h₁ : 0 < ‖M i j‖ := by rwa [norm_pos_iff]
+    have h₂ : 0 < ‖M i j‖ ^ 3 := by positivity
+    rwa [← mul_le_mul_left h₂]
+
+open CStarModule in
+lemma norm_le_of_forall_inner_le [DecidableEq n] {M : CStarMatrix m n A} {C : ℝ≥0}
+    (h : ∀ v w, ‖⟪w, toCLM A M v⟫_A‖ ≤ C * ‖v‖ * ‖w‖) : ‖M‖ ≤ C := by
+  let instNACG : NormedAddCommGroup (CStarMatrix m n A) := NormedAddCommGroup.ofCore normedSpaceCore
+  change ‖toCLM A M‖ ≤ C
+  rw [ContinuousLinearMap.opNorm_le_iff NNReal.zero_le_coe]
+  intro v
+  rw [norm_eq_csSup]
+  refine (csSup_le_iff ?bddAbove ?nonempty).mpr ?bound
+  case bddAbove =>
+    refine ⟨‖M‖ * ‖v‖, ?_⟩
+    rw [mem_upperBounds]
+    intro b hb
+    obtain ⟨w, hw₁, hw₂⟩ := hb
+    rw [← hw₂]
+    calc _ ≤ ‖w‖ * ‖toCLM A M v‖ := norm_inner_le (m →C⋆ A)
+      _ ≤ ‖w‖ * (‖M‖ * ‖v‖) := by
+            gcongr
+            exact ContinuousLinearMap.le_opNorm ((toCLM A) M) v
+      _ ≤ 1 * (‖M‖ * ‖v‖) := by gcongr
+      _ = ‖M‖ * ‖v‖ := by simp
+  case nonempty => exact ⟨0, 0, by simp, by simp⟩
+  case bound =>
+    intro b hb
+    obtain ⟨w, hw₁, hw₂⟩ := hb
+    rw [← hw₂]
+    calc _ ≤ C * ‖v‖ * ‖w‖ := h v w
+      _ ≤ C * ‖v‖ * 1 := by gcongr
+      _ = C * ‖v‖ := by simp
```

Comment:
This one: `lemma x = 0 ∨ 0 < ‖x‖` for `NormedAddCommGroup`.

I want to be able to write `rfl` in the first argument of `obtain`. I know it's minor, but I've definitely run into this before, and this time it clicked because I did it twice back-to-back.

Reply:
I just added this lemma.